yarn-project/simulator/src/public/executor.ts

```diff
@@ -1,6 +1,9 @@
 import { GlobalVariables, Header, PublicCircuitPublicInputs } from '@aztec/circuits.js';
 import { createDebugLogger } from '@aztec/foundation/log';
 
+import { spawn } from 'child_process';
```

Comment:
I've never used spawn in TS, but a few pointers

* Might be a problem in Windows: https://www.npmjs.com/package/cross-spawn . I guess we don't care because we are already using spawn?
* This package seems to be used somewhere else in the code: https://github.com/jamiebuilds/spawndamnit , no idea what it does.

Reply:
I think the package does essentially what im doing by promisifying the spawn. I'll have a look how useful this is though
> Automatically kills all spawn processes